compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6190,9 +6190,9 @@ private BLangRecordLiteral.BLangRecordKeyValueField createRecordKeyValueField(Lo
     }
 
     public void generateFieldsForUserUnspecifiedRecordFields(BLangRecordLiteral recordLiteral,
-                                                              List<RecordLiteralNode.RecordField> userSpecifiedFields) {
+                                                             List<RecordLiteralNode.RecordField> userSpecifiedFields) {
         BType type = Types.getImpliedType(recordLiteral.getBType());
-        if (type.getKind() != TypeKind.RECORD) {
+        if (type.getKind() != TypeKind.RECORD || isSpreadingAnOpenRecord(userSpecifiedFields)) {
```

Comment:
Why do we need this check again?

Reply:
We call this method for any `BLangRecordLiteral` [here](https://github.com/ballerina-platform/ballerina-lang/blob/bc7e5e948b6dadc96ea5c781c751904e98f3f3fc/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java#L6088), so we need to determine if the compiler need to explicitly add fields to the RecordLiteral or not in this method.

Reply:
Added comment explaining why we need this